src/libstd/sys/unix/os.rs

```diff
@@ -483,7 +483,6 @@ pub fn home_dir() -> Option<PathBuf> {
                   target_os = "nacl",
                   target_os = "emscripten")))]
     unsafe fn fallback() -> Option<OsString> {
-        #[cfg(not(target_os = "solaris"))]
         unsafe fn getpwduid_r(me: libc::uid_t, passwd: &mut libc::passwd,
```

Comment:
Ah, you're right.  I was attempting to minimise changes here, but the wrapper could be removed entirely.  Let me simplify this further and rebuild.